### DISCUSSION TITLE
mount external volume on (read-only) /usr

### DISCUSSION THREAD

**Discussion by till** [flatcar/Flatcar#1206] [Q&A] [answered]
Is it possible to (bind) mount a volume to a path on the read-only `/usr` partition? 

As in, my k8s distro wants to write to `/usr/libexec/k0s`. All my attempts to mount anything there, result in mount failures all over the place when the node boots (and ran ignition). But I was curious if that is somehow possible.

**Comment by pothos** [comment, marked as answer]
A bind mount may be lost due to systemd-sysext images being (re)loaded. A solution that brings in a symlink through a systemd-sysext image is what we did here: https://github.com/flatcar/sysext-bakery/blob/7bbe7864bac89cbbeb2f71f206a794240bd63155/create_kubernetes_sysext.sh#L87

We should add this symlink to the generic image, I created an issue for that: https://github.com/flatcar/Flatcar/issues/1193

**Reply by pothos** [reply]
I created a PR https://github.com/flatcar/scripts/pull/1238

**Reply by till** [reply]
Is #1328 related?

**Reply by pothos** [reply]
Not really, the libexec symlink is part of the real `/usr`, not an overlay.
